crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -483,21 +447,20 @@ impl SettlementEncoder {
         self.tokens.sort();
 
         for i in 0..self.trades.len() {
-            let sell_token_index = self
-                .token_index(self.trades[i].data.order.data.sell_token)
-                .expect("missing sell token for existing trade");
-
             self.trades[i].tokens = match self.trades[i].tokens {
                 TokenReference::Indexed { .. } => TokenReference::Indexed {
-                    sell_token_index,
+                    sell_token_index: self
+                        .token_index(self.trades[i].data.order.data.sell_token)
+                        .expect("missing sell token for existing trade"),
                     buy_token_index: self
                         .token_index(self.trades[i].data.order.data.buy_token)
                         .expect("missing buy token for existing trade"),
                 },
                 TokenReference::CustomPrice {
```

Comment:
Instead of destructuring the original value and recreating it identically you should be able to do sth like 👇 in the match statement:
```rust
original @ TokenReference::CustomPrice => original
```

Reply:
I think it is not applicable here since I am not returning the original value but converting to the tuple (U256, U256).

Reply:
The conversion to a tuple happens somewhere else. In this case you return a `TokenReference` again. Just to double check I tried to compile my suggestion and it works.

Reply:
Ahhh yes, was looking at the other line of the code :)